File: Programming Basics with Python/Проверки/lunch_break.py
import math

series_name = input()
ep_time = int(input())
rest = int(input())

lunch = rest * 1/8
relaxation = rest * 1/4
remaining_time = rest - (lunch + relaxation)

if remaining_time >= ep_time:
    left_time = math.ceil(remaining_time - ep_time)
    print(f"You have enough time to watch {series_name} and left with {left_time} minutes free time.")
else:
    needed_time = math.ceil(ep_time - remaining_time)
    print(f"You don't have enough time to watch {series_name}, you need {needed_time} more minutes.")


"""
from math import ceil

series_name = input()
series_duration = int(input())
lunch_break_duration = int(input())

lunch_duration = lunch_break_duration / 8
recreation_time = lunch_break_duration / 4

time_for_watching_series = lunch_break_duration - (lunch_duration + recreation_time)

if time_for_watching_series >= series_duration:
    print(
        f"You have enough time to watch {series_name} and left with {ceil(time_for_watching_series - series_duration)} "
        f"minutes free time.")
else:
    print(
        f"You don't have enough time to watch {series_name}, you need {ceil(series_duration - time_for_watching_series)} "
        f"more minutes.")
"""

File: Programming Basics with Java/Checks/LunchBreak.java
package Checks;
import java.util.Scanner;

public class LunchBreak 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        String tv_show = input.nextLine();
        int ep = input.nextInt();
        int time_off = input.nextInt();

        double lunch_time = time_off / 8.0;
        double break_time = time_off / 4.0;
        double time_left = time_off - lunch_time - break_time;
        double diff = Math.ceil(ep - time_left);

        if (diff > 0)
        {
            System.out.printf("You don't have enough time to watch %s, you need %.0f more minutes.", tv_show, Math.abs(diff));
        }
        else
        {
            System.out.printf("You have enough time to watch %s and left with %.0f minutes free time.", tv_show, diff);
        }
    }
}
